pkg/collector/corechecks/system/disk/disk/disk.go

```diff
@@ -2,138 +2,252 @@
 
-	// all good, don't exclude the disk
-	return false
+	return nil
 }
 
-func (c *Check) instanceConfigure(data integration.Data) error {
-	conf := make(map[interface{}]interface{})
-	c.cfg = &diskConfig{}
-	err := yaml.Unmarshal([]byte(data), &conf)
+func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, data integration.Data, initConfig integration.Data, source string) error {
+	err := c.CommonConfigure(senderManager, initConfig, data, source)
 	if err != nil {
 		return err
 	}
+	return c.diskConfigure(data, initConfig)
+}
 
-	useMount, found := conf["use_mount"]
-	if useMount, ok := useMount.(bool); found && ok {
-		c.cfg.useMount = useMount
+func (c *Check) diskConfigure(data integration.Data, initConfig integration.Data) error {
+	unmarshalledInstanceConfig := make(map[interface{}]interface{})
+	err := yaml.Unmarshal([]byte(data), &unmarshalledInstanceConfig)
+	if err != nil {
+		return err
 	}
-
-	excludedFilesystems, found := conf["excluded_filesystems"]
-	if excludedFilesystems, ok := excludedFilesystems.([]string); found && ok {
-		c.cfg.excludedFilesystems = excludedFilesystems
+	unmarshalledInitConfig := make(map[interface{}]interface{})
+	err = yaml.Unmarshal([]byte(initConfig), &unmarshalledInitConfig)
+	if err != nil {
+		return err
 	}
 
-	// Force exclusion of CDROM (iso9660) from disk check
-	c.cfg.excludedFilesystems = append(c.cfg.excludedFilesystems, "iso9660")
-
-	excludedDisks, found := conf["excluded_disks"]
-	if excludedDisks, ok := excludedDisks.([]string); found && ok {
-		c.cfg.excludedDisks = excludedDisks
+	deprecationsInitConf := map[string]string{
+		"file_system_global_blacklist": "file_system_global_exclude",
+		"device_global_blacklist":      "device_global_exclude",
+		"mount_point_global_blacklist": "mount_point_global_exclude",
 	}
-
-	excludedDiskRe, found := conf["excluded_disk_re"]
-	if excludedDiskRe, ok := excludedDiskRe.(string); found && ok {
-		c.cfg.excludedDiskRe, err = regexp.Compile(excludedDiskRe)
-		if err != nil {
-			return err
+	for oldKey, newKey := range deprecationsInitConf {
+		if _, exists := unmarshalledInitConfig[oldKey]; exists {
+			log.Warnf("`%s` is deprecated and will be removed in a future release. Please use `%s` instead.", oldKey, newKey)
 		}
 	}
 
-	tagByFilesystem, found := conf["tag_by_filesystem"]
-	if tagByFilesystem, ok := tagByFilesystem.(bool); found && ok {
-		c.cfg.tagByFilesystem = tagByFilesystem
+	deprecationsInstanceConf := map[string]string{
+		"file_system_whitelist":  "file_system_include",
+		"file_system_blacklist":  "file_system_exclude",
+		"device_whitelist":       "device_include",
+		"device_blacklist":       "device_exclude",
+		"mount_point_whitelist":  "mount_point_include",
+		"mount_point_blacklist":  "mount_point_exclude",
+		"excluded_filesystems":   "file_system_exclude",
+		"excluded_disks":         "device_exclude",
+		"excluded_disk_re":       "device_exclude",
+		"excluded_mountpoint_re": "mount_point_exclude",
```

Comment:
Same as with `deprecationsInitConf` should we move it to a constant?

Reply:
Same comment as above